### ISSUE TITLE
Bogus NotWritableError during "conda create"

### ISSUE THREAD

**Issue by whamblen** [conda/conda#8335]
## Current Behavior

In a home directory where the user has read/write permissions *and* .conda doesn't already exist *and* the user isn't allowed to set the sgid bit on .conda then "conda create -n mynewenv" will fail with "NotWritableError: The current user does not have write permissions to a required path."  

### Steps to Reproduce

```
The unusual element here is the user not having permission to do "chmod 02755 .conda".  "conda create" tries to do this immediately after "mkdir .conda".  In our case the back end storage for the NFS home directories is an Isilon and it is probably something specific to that which prevents the chmod command from working.  But there are other ways this could happen.

sgid can be essential for shared environments but is also completely unnecessary in many common use cases so I think simply aborting here is a bug.
```

## Expected Behavior

"conda create -n mynewnev" should succeed even if the sgid bit cannot be set on a newly created .conda (or .conda/envs and .conda/pkgs which exhibit the same behavior).   Depending on why the sgid is being attempted it may be worth throwing up an warning with a chance to abort if the use case was going to require sgid.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
    active environment : None
            shell level : 0
       user config file : /dartfs-hpc/rc/home/9/d1070v9/.condarc
 populated config files : 
          conda version : 4.6.3
    conda-build version : not installed
         python version : 3.7.1.final.0
       base environment : /dartfs-hpc/rc/lab/R/RCStaff/miniconda/miniconda  (read only)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/linux-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /dartfs-hpc/rc/lab/R/RCStaff/miniconda/miniconda/pkgs
                          /dartfs-hpc/rc/home/9/d1070v9/.conda/pkgs
       envs directories : /dartfs-hpc/rc/home/9/d1070v9/.conda/envs
                          /dartfs-hpc/rc/lab/R/RCStaff/miniconda/miniconda/envs
               platform : linux-64
             user-agent : conda/4.6.3 requests/2.21.0 CPython/3.7.1 Linux/3.10.0-862.9.1.el7.x86_64 rhel/7.6 glibc/2.17
                UID:GID : 20335:480987
             netrc file : None
           offline mode : False

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
Empty output
```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at /dartfs-hpc/rc/lab/R/RCStaff/miniconda/miniconda:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py37_0    defaults
ca-certificates           2019.1.23                     0    defaults
certifi                   2018.11.29               py37_0    defaults
cffi                      1.11.5           py37he75722e_1    defaults
chardet                   3.0.4                    py37_1    defaults
conda                     4.6.3                    py37_0    defaults
conda-env                 2.6.0                         1    defaults
cryptography              2.5              py37h1ba5d50_0    defaults
idna                      2.8                      py37_0    defaults
libedit                   3.1.20181209         hc058e9b_0    defaults
libffi                    3.2.1                hd88cf55_4    defaults
libgcc-ng                 8.2.0                hdf63c60_1    defaults
libstdcxx-ng              8.2.0                hdf63c60_1    defaults
ncurses                   6.1                  he6710b0_1    defaults
openssl                   1.1.1a               h7b6447c_0    defaults
pip                       18.1                     py37_0    defaults
pycosat                   0.6.3            py37h14c3975_0    defaults
pycparser                 2.19                     py37_0    defaults
pyopenssl                 18.0.0                   py37_0    defaults
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.1                h0371630_7    defaults
readline                  7.0                  h7b6447c_5    defaults
requests                  2.21.0                   py37_0    defaults
ruamel_yaml               0.15.46          py37h14c3975_0    defaults
setuptools                40.6.3                   py37_0    defaults
six                       1.12.0                   py37_0    defaults
sqlite                    3.26.0               h7b6447c_0    defaults
tk                        8.6.8                hbc83047_0    defaults
urllib3                   1.24.1                   py37_0    defaults
wheel                     0.32.3                   py37_0    defaults
xz                        5.2.4                h14c3975_4    defaults
yaml                      0.1.7                had09818_2    defaults
zlib                      1.2.11               h7b6447c_3    defaults
```
</p></details>

**Comment by jjhelmus** [comment]
This issue was fixed by #8253 which is included in conda 4.6.4 and above.  Upgrading conda should fix the  problem.

**Comment by jjhelmus** [comment]
I'm less sure that this was fixed by #8253.  @whamblen can you upgrade conda and see if this issue persists?

**Comment by whamblen** [comment]
Sorry about that.  It does do the same thing with conda 4.6.7.  I just double checked to be certain.   I had downgraded conda while investigating a different issue that I thought I remembered working with 4.6.3. That is what the "conda info" output in the report was from.

Anyway, because the directory creation does succeed (and the chmod isn't attempted if it's there) if you try and fail enough times, conda create will work.  Equivalently, you can make the missing directories before running conda create.  e.g. this will work around it.

$ mkdir -p .conda/pkgs/cache .conda/envs
